infrastructure/modules/aks/monitor.tf

```
@@ -0,0 +1,39 @@
+resource "azurerm_log_analytics_workspace" "aks" {
+  name                = var.azurerm_log_analytics_workspace_aks_name != "" ? var.azurerm_log_analytics_workspace_aks_name : "${var.prefix}-${var.environment}-aks-law"
+  resource_group_name = azurerm_resource_group.monitor.name
+  location            = azurerm_resource_group.monitor.location
+  retention_in_days   = 30
+  identity {
+    type = "SystemAssigned"
+  }
+}
+
+resource "azurerm_monitor_workspace" "aks" {
+  name                = var.azurerm_monitor_workspace_aks_name != "" ? var.azurerm_monitor_workspace_aks_name : "${var.prefix}-${var.environment}-aks-amw"
+  resource_group_name = azurerm_resource_group.monitor.name
+  location            = azurerm_resource_group.monitor.location
+}
+
+resource "random_id" "aks" {
+  byte_length = 3 # 3 gives 6 characters
+}
+
+resource "azurerm_storage_account" "aks" {
+  name                     = var.azurerm_storage_account_aks_name != "" ? var.azurerm_storage_account_aks_name : "${var.prefix}${var.environment}akslog${random_id.aks.hex}"
+  resource_group_name      = azurerm_resource_group.monitor.name
+  location                 = azurerm_resource_group.monitor.location
+  account_tier             = "Standard"
+  account_replication_type = "ZRS"
+  account_kind             = "StorageV2"
+  min_tls_version          = "TLS1_2"
+  is_hns_enabled           = true
```

We should disallow creation of public containers and disable access keys if possible

```suggestion
  is_hns_enabled           = true
  allow_nested_items_to_be_public = false
  shared_access_key_enabled = false
```